### ISSUE TITLE
Clarification regarding model

### ISSUE THREAD

**Issue by Mohamedkaif10** [LIYUESEN/druggpt#34]
I have a few doubts that I couldn't understand properly:

1)You have used two models, GPT-2 and GPT2Head model. Why is that? Which one is more important and contributes more?
2)Have you fine-tuned both models with the dataset, or only one?
3)Have you trained using both the ZINC20 and JGLaser datasets, or only one?

**Comment by LIYUESEN** [comment]
Dear Mohamedkaif10

1. We selected the GPT2LMHeadModel from the Hugging Face Transformers library as our base model. GPT2LMHeadModel extends GPT2Model by adding a language modeling head, which maps hidden state vectors to the vocabulary size, enabling autoregressive text generation. More details can be found in the https://huggingface.co/docs/transformers/model_doc/gpt2#transformers.GPT2LMHeadModel

2. Our tokenizer and model are both trained from scratch. 

3. In this study, we utilized two datasets to train the DrugGPT model. 
    https://huggingface.co/datasets/zpn/zinc20
    https://huggingface.co/datasets/jglaser/binding_affinity 

More details will be described in our upcoming paper. Stay tuned!

Best,
Yuesen Li

**Comment by Mohamedkaif10** [comment]
Thank you so much for clarification, 
Really excited and waiting for your next paper Yuesen Li

 Best,
Mohamedkaif

**Comment by Mohamedkaif10** [comment]
Dear Yuesen Li
I have trained my model using jglaser, but when I attempted to evaluate it with no_kras, I noticed that many sequences in the test set overlap with those in the training set. Given this issue, I wanted to understand your approach to evaluating the model for jglaser.

Did you evaluate it by splitting the training dataset into specific percentages, or did you use a different method for testing?

**Comment by LIYUESEN** [comment]
Split the training set at a 9:1 ratio.

**Comment by Mohamedkaif10** [comment]
Thank you..

**Comment by Mohamedkaif10** [comment]
Also in the paper you have not explained the reinforcement learning , but in the code of before you have used a slight Reinforcement leaning , have u used it or its just additional thing, can u clarify about it.

**Comment by Mohamedkaif10** [comment]
I have one doubt as the paper says 

"Further, we analyzed ligands from the jglaser/binding_affinity database that had already bound with proteins. After extracting the information of these ligands, we applied the BPE algorithm for tokenization. The results showed that 3560 tokens were sufficient to accurately describe these ligands that had already bound with proteins."

I also used BPE algorithm for tokenizing the ligands in the jglaser but im not getting 3560 tokens , im getting in few hunderads , 
so what can i do

**Comment by Mohamedkaif10** [comment]
hello 
I found that there is error in my num of merges , can u say how many merges have u used , while using the bpe tokenizer

**Comment by Mohamedkaif10** [comment]
Can u share the algorithm u have used for the BPE tokenizer and also the refering material (like code) for the BPE tokenizer if u can please??

**Comment by LIYUESEN** [comment]
https://huggingface.co/docs/transformers/main/en/main_classes/tokenizer#transformers.PreTrainedTokenizerFast.train_new_from_iterator

https://github.com/huggingface/transformers/blob/main/src/transformers/tokenization_utils_fast.py#L730

**Comment by Mohamedkaif10** [comment]
Thanks Liyuesen
